### PR TITLE
`Learning paths`: Add learning path completion button

### PR DESCRIPTION
**Pull request by JohannesWt**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Currently students don't really know when their learning path is (temporarily) completed.

### Description

This PR introduces a new button which indicates that the learning path has been (temporarily) completed. Additionally when clicking on the completion button the system now displays a short description that the learning path has been completed but in new units could be added in the near future. Meanwhile students still have the opportunity to recap the already completed learning objects.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- at least 1 lecture
- at least 1 competency with at least 1 video unit, text unit or similar

1. Log in to Artemis
2. Navigate to the learning paths UI
3. Finish the learning path
4. Verify that you see the completion button
5. Click the completion button
6. Verify that you see the Congratulations text
7. Verify that you still can open the navigation overview and navigate to completed learning objects

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

unchanged due to small additions

### Screenshots

image
image

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Complete" button to mark learning paths as finished, enhancing user interaction.
	- Added conditional logic to improve the display of learning objects and navigation options based on their availability.
	- Implemented enhanced loading state handling for a clearer user experience during data retrieval.
	- Updated navigation structure to include congratulatory messages upon completing learning paths.
	- Enhanced German localization with a new "completeButton" and improved labels for clarity and user engagement.

- **Bug Fixes**
	- Improved navigation logic to handle cases when no current learning object is found, providing a more informative response.

- **Localization**
	- Updated English and German localization files to include new buttons and messages for improved user feedback and navigation.

- **Tests**
	- Enhanced test cases for the `LearningPathNavigationService` to validate the completion functionality and improve type safety.

### CODE REVIEW

File: src/main/webapp/i18n/de/learningPath.json
Comment (rabeatwork) [inline src/main/webapp/i18n/de/learningPath.json:14]: Lernpfadobjekte, is there a better word? 🤔

**Review by rabeatwork — CHANGES_REQUESTED** [review]

**Review by pzdr7 — APPROVED** [review]
Tested on TS3. LGTM :+1:

**Review by JohannesStoehr — APPROVED** [review]
Code

**Review by N0W0RK — DISMISSED** [review]
Code LGTM

**Review by sarpsahinalp — CHANGES_REQUESTED** [review]
image
Horizontal and vertical overflow and scrollable is this intended?

**Review by az108 — COMMENTED** [review]
except @sarpsahinalp comment, works as expected. Some size adjustments are needed to get rid of the appearing scrollbars. Tested on TS1
image

File: src/main/webapp/i18n/de/learningPath.json
Comment (JohannesWt) [reply src/main/webapp/i18n/de/learningPath.json]: Done 🇹🇭

**Review by sarpsahinalp — DISMISSED** [review]
Altough the last lecture is not completed the complete button is clickable it should be disabled:
image

**Review by pzdr7 — COMMENTED** [review]
Re-tested on TS1. When I reach the end of the learning path without all units completed, the button to complete it is available and leads to the start (?) of the path.

**Review by az108 — APPROVED** [review]
Tested on TS1, looks good to me after changes

**Review by JohannesStoehr — DISMISSED** [review]
Reapprove code

**Review by sarpsahinalp — DISMISSED** [review]
Approve after changes, works as expected tested on TS1.

**Review by pzdr7 — DISMISSED** [review]
Tested on TS1, thanks for fixing the looping issue 👍

File: src/main/webapp/i18n/de/learningPath.json
Comment (rabeatwork) [inline src/main/webapp/i18n/de/learningPath.json:11]: Lernheiten? 🤓

**Review by rabeatwork — CHANGES_REQUESTED** [review]

**Review by JohannesStoehr — APPROVED** [review]
Reapprove code

**Review by az108 — APPROVED** [review]
reapprove after typo

**Review by pzdr7 — APPROVED** [review]
Reapprove

**Review by N0W0RK — APPROVED** [review]
Re approve

**Review by rabeatwork — APPROVED** [review]
maintainer approve

**Review by MarkusPaulsen — APPROVED** [review]
Tested on TS3 + Code looks good

**Review by MaximilianAnzinger — APPROVED** [review]

**Pull request merged**
